[#187364742] - Feature(Product Listing): A User Should be able to list all Items

AngeloChristian1 opened the pull request:
**What does this PR do?**
This PR introduces functionality that allows users to view products, sellers will see the items in their collection and buyers will be able to see all items available


**Description of Task to be completed?**
  1. A seller should be able to retrieve a list of all items in their collection
  2. A buyer should be able to retrieve a list of all items on the available list of products

#### How should this be manually tested?
1. Start the application server
2. Use Swagger or Postman to test relevant endpoints
3. Login as a buyer or as a seller
4. Navigate to `collection/products`
5. If the user is a buyer all available products will be displayed
6. If user is a seller only products in their collection will be displayed
7. Also user can display one product by navigating to `collection/products/:productId` and the same considerations apply

#### Any background context you want to provide?
No additional context, configurations or packages were added expept moment for time comparison
After pulling make sure you install `moment`

**Screenshot or Video**
https://www.loom.com/share/1a6b0df6302f41348bfbb0ef47d54cfa?sid=dcc48902-9a1d-471b-b7dd-5a2056be3948
Pivotal Tracker Story
#187364742

Bananayosostene commented on src/routes/productRoute.ts:
```diff
@@ -22,7 +22,16 @@ router.post(
 router.post(
   '/collection',
   isAuthenticated,
-  checkPermission(UserRole.SELLER), validateRequest(collectionSchema, 'body'),
```
I can see this line was deleted, why

Bananayosostene commented on src/services/productServices.ts:
```diff
@@ -0,0 +1,29 @@
+import Collection from "../database/models/collectionModel"
+
+
+// eslint-disable-next-line import/prefer-default-export
```
remove eslint disable it can be better to user proper way

AngeloChristian1 committed:
Feature (Product Listing): A user will be able to list products

1. Added functionality for a buyer to view all available products
2. Added functionality for admin to view
2. Added functionality for a seller to view products in a collection they own

  Delivers [#187364742]

Furaha-Justine reviewed (approved):
Great work.

mpanoathos committed:
feat(changeStatus):Admin should change status of a user (#52)

[Delivers #187364736]
